test/unit/cmap/connection.test.ts

```diff
@@ -31,10 +31,15 @@ const connectionOptionsDefaults = {
   loadBalanced: false
 };
 
-/** The absolute minimum socket API needed by Connection as of writing this test */
+/**
```

Comment:
Where are the tests ensuring we unset the socket timeout on the connection?

Reply:
https://github.com/mongodb/node-mongodb-native/blob/NODE-4854-minPoolSizeReconnections/test/unit/cmap/connect.test.ts#L123-L246

Reply:
Can we move those tests to `connection.test.ts`?  They're not connect related.